### PR TITLE
Generate and pass new member key share to CCF

### PR DESCRIPTION
**Pull request by jumaffre**
Addresses 2. in https://github.com/microsoft/CCF/issues/51

Each member now owns `ksk_m`, a private/public key pair (Ed25519, so we can later do asymmetric encryption with [Evercrypt](https://hacl-star.github.io/HaclNaCl.html#public-key-encryption-box). The public key pair is recorded in the `ccf.members` table in CCF and will be (next PR) used to encrypt each member's share.

This keypair is generated via the `keygenerator.sh` script (`--gen-key-share` option) and passed to the `cchost` CLI along with the member certificate, e.g. `cchost ... start --member-info=member0_cert.pem,member0_kshare_pub.pem`.

### CODE REVIEW

File: src/ds/cli_helper.h [hunk 1]
```diff
@@ -58,4 +63,61 @@ namespace cli
...
+        }
+
+        parsed.emplace_back(cert, keyshare_pub);
+      }
+      return true;
+    };
+
+    auto* option = app.add_option(option_name, fun, option_desc, true);
+    option->type_name("member_cert.pem,member_kshare_pub.pem.pem")
+      ->type_size(-1);
```
Comment (jumaffre) [inline src/ds/cli_helper.h]: FYI, `type_size(-1)` tells CLI11 that the option is a vector.

**Comment by cimetrics** [comment]
add_member_encryption_key@[4814 aka 20200211.30](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4814&view=results) vs master ewma over 30 builds from [4447](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4447&view=results) to [4808](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4808&view=results)
images

File: sphinx/source/members/index.rst [hunk 2]
```diff
@@ -9,18 +9,21 @@ The ``CCF/tests/keygenerator.sh`` script can be used to generate the member's ce
...
+    Key share private key generated at:  member1_kshare_pub.pem
 
 .. note:: See :ref:`developers/cryptography:Algorithms and Curves` for the list of supported cryptographic curves.
 
-The member's private key (e.g. ``member1_privk.pem``) should be stored on a trusted device while the certificate (e.g. ``member1_cert.pem``) should be given to operators before starting the first node of a new CCF network.
+The member's private key (e.g. ``member1_privk.pem``) should be stored on a trusted device while the certificate (e.g. ``member1_cert.pem``) and public key share (e.g. ``member1_kshare_pub.pem``) should be given to operators before starting the first node of a new CCF network.
 
 
-.. toctree::
+.. toctree::d
```
Comment (eddyashton) [inline sphinx/source/members/index.rst]: What does this `d` ddo?

Comment (jumaffre) [reply sphinx/source/members/index.rst]: Woops!

File: src/apps/luageneric/test/luageneric_test.cpp [hunk 3]
```diff
@@ -80,6 +80,7 @@ void set_lua_logger()
 
 auto user_caller = kp -> self_sign("CN=name");
 auto user_caller_der = tls::make_verifier(user_caller) -> der_cert_data();
+auto dummy_key_share = std::vector<uint8_t>(32);
```
Comment (achamayou) [inline src/apps/luageneric/test/luageneric_test.cpp]: 32?

File: src/ds/cli_helper.h [hunk 4]
```diff
@@ -40,11 +43,13 @@ namespace cli
...
       {
-        throw std::logic_error("Port for " + option_name + " is not a number");
+        throw CLI::ValidationError(option_name, "Port is not a number");
       }
       if (port_int <= 0 || port_int > 65535)
-        throw std::logic_error(
-          "Port number is not in range 1-65535 for " + option_name);
+      {
+        throw CLI::ValidationError(
+          option_name, "Port number is not in range 1-65535");
```
Comment (achamayou) [inline src/ds/cli_helper.h]: 0-65535

File: src/ds/cli_helper.h (hunk 1 above)
Comment (achamayou) [reply src/ds/cli_helper.h]: .pem.pem?

File: src/host/main.cpp [hunk 5]
```diff
@@ -224,14 +224,12 @@ int main(int argc, char** argv)
...
-      member_cert_files,
-      "Certificate files of the initial consortium members",
-      true)
-    ->check(CLI::ExistingFile)
+  std::vector<cli::ParsedMemberInfo> members_info;
+  cli::add_member_info_option(
+    *start,
+    members_info,
+    "--member-info",
+    "Initial consortium members information (public identity,public key share")
```
Comment (achamayou) [inline src/host/main.cpp]: Unclosed, missing space?

**Pull request merged**